packages/search-input/src/SearchInput/SearchInput.tsx

```diff
@@ -0,0 +1,408 @@
+import React, {
+  ChangeEvent,
+  EventHandler,
+  FocusEventHandler,
+  FormEventHandler,
+  KeyboardEventHandler,
+  MouseEventHandler,
+  SyntheticEvent,
+  useCallback,
+  useMemo,
+  useRef,
+  useState,
+} from 'react';
+import isUndefined from 'lodash/isUndefined';
+
+import { cx } from '@leafygreen-ui/emotion';
+import {
+  useBackdropClick,
+  useControlledValue,
+  useDynamicRefs,
+  useForwardedRef,
+} from '@leafygreen-ui/hooks';
+import MagnifyingGlass from '@leafygreen-ui/icon/dist/MagnifyingGlass';
+import XIcon from '@leafygreen-ui/icon/dist/XWithCircle';
+import IconButton from '@leafygreen-ui/icon-button';
+import LeafyGreenProvider, {
+  useDarkMode,
+} from '@leafygreen-ui/leafygreen-provider';
+import {
+  getNodeTextContent,
+  isComponentType,
+  keyMap,
+} from '@leafygreen-ui/lib';
+
+import { SearchInputContextProvider } from '../SearchInputContext';
+import { SearchResultsMenu } from '../SearchResultsMenu';
+import { convertEvent } from '../utils/convertEvent';
+
+import {
+  baseInputStyle,
+  clearButtonSizeStyle,
+  formStyle,
+  inputThemeStyle,
+  inputWrapperDisabledStyle,
+  inputWrapperFocusStyles,
+  inputWrapperInteractiveStyles,
+  inputWrapperSizeStyle,
+  inputWrapperStyle,
+  inputWrapperThemeStyle,
+  searchIconDisabledStyle,
+  searchIconSizeStyle,
+  searchIconThemeStyle,
+} from './SearchInput.styles';
+import { SearchInputProps, Size, State } from './SearchInput.types';
+
+/**
+ * # SearchInput
+ *
+ * SearchInput component
+ *
+ * @param props.state The current state of the SearchInput. This can be none, or loading.
+ * @param props.darkMode determines whether or not the component appears in dark theme.
+ * @param props.size determines the size of the text and the height of the input.
+ * @param props.baseFontSize determines the base font size if size is set to default.
+ */
+
+export const SearchInput = React.forwardRef<HTMLInputElement, SearchInputProps>(
+  function SearchInput(
+    {
+      placeholder = 'Search',
+      className,
+      darkMode: darkModeProp,
+      size = Size.Default,
+      disabled,
+      children,
+      state = State.Unset,
+      value: valueProp,
+      onChange: onChangeProp,
+      onSubmit: onSubmitProp,
+      'aria-label': ariaLabel,
+      'aria-labelledby': ariaLabelledBy,
+      ...rest
+    }: SearchInputProps,
+    forwardRef: React.Ref<HTMLInputElement>,
+  ) {
+    const { theme, darkMode } = useDarkMode(darkModeProp);
+    const [isOpen, setOpen] = useState(false);
+    // The index of the currently highlighted result option
+    const [highlightIndex, setHighlightIndex] = useState<number>(0);
+
+    const closeMenu = () => setOpen(false);
+    const openMenu = () => setOpen(true);
+
+    const formRef = useRef<HTMLFormElement>(null);
+    const searchBoxRef = useRef<HTMLDivElement>(null);
+    const menuRef = useRef<HTMLUListElement>(null);
+    const inputRef = useForwardedRef(forwardRef, null);
```

Comment:
does useForwardRef allow you to both use the ref and pass it to the forward ref?

Reply:
Exactly, yeah